docs/event-bus/docs/013-service-programming-model.md

```diff
@@ -0,0 +1,76 @@
+| x-b3-spanid | This header is used by the Zipkin tracer in Envoy. The SpanId is 64-bit in length and indicates the position of the current operation in the trace tree. The value should not be interpreted: it may or may not be derived from the value of the TraceId. See more on zipkin tracing [here](https://github.com/openzipkin/b3-propagation). e.g. `7350c92dc6fa825e` |
+|x-b3-traceid | This header is used by the Zipkin tracer in Envoy. The TraceId is 64-bit in length and indicates the overall ID of the trace. Every span in a trace shares this ID. See more on zipkin tracing [here](https://github.com/openzipkin/b3-propagation). e.g. `02c04beed9aa6c53` |
+| x-request-id | Randomly generated an ID for identifies the HTTP request delivering the business event. e.g. `ec0ffa2a-13b6-9e63-9563-12e5c0e27e38` |
+| x-envoy-decorator-operation | If this header is present on ingress requests, its value will override any locally defined operation (span) name on the server span generated by the tracing mechanism. Similarly, if this header is present on an egress response, its value will override any locally defined operation (span) name on the client span. e.g. `default-route` |
+| x-envoy-expected-rq-timeout-ms | This is the time in milliseconds the router expects the request to be completed. Envoy sets this header so that the upstream host receiving the request can make decisions based on the request timeout, e.g., early exit. This is set on internal requests and is either taken from the `x-envoy-upstream-rq-timeout-ms` header or the route timeout, in that order. e.g. `15000` |
+| x-istio-attributes | istio specific metadata. e.g. `CkMKCnNvdXJjZS51aWQSNRIza3ViZXJuZXRlczovL2NvcmUtcHVzaC01ZDg5OTU1ZjhiLXhjZ3ZyLmt5bWEtc3lzdGVtCh8KCXNvdXJjZS5pcBISMhAAAAAAAAAAAAAA//+sEQAb` |
+
+## Event Payload
+
+The event payload is delivered as the body of the HTTP Request in JSON format. The JSON schema is available in the Service Catalog in the registered service for the remote events.
+
+## Event Payload Example
+
+For example, writing a service interested in an event `order.created` published by `ec` service. the published event is schema is something like:
+
+```json
+{
+  "example": {
+    "orderCode": "4caad296-e0c5-491e-98ac-0ed118f9474e"
+  },
+  "properties": {
+    "orderCode": {
+      "description": "Resource identifier",
+      "title": "OrderCode",
+      "type": "string"
+    }
+  },
+  "type": "object"
+}
+```
+
+So, the HTTP POST request payload would be a JSON object like:
+
+```json
+{"orderCode": "4caad296-e0c5-491e-98ac-0ed118f9474e"}
+```
+
+## Successful Delivery
+
+The delivered message to a subscriber would be considered as successfully consumed if the service HTTP response status code was `2xx`. However, responding with a status code, not a `2xx` (< 200 or >= 300) means an unsuccessful message consumption and the message delivery will be retried. And this implies **At-least-once** delivery guarantee.
```

Comment:
``A message delivered to a subscriber is considered successfully consumed if the service's HTTP response status code is `2xx`. If the status code is not `2xx` (< 200 or >= 300), it means that a message consumption is not successful and that the message delivery is re-tried. This implies **At-least-once** delivery guarantee.``